pkg/cmd/readyz/readyz.go

```diff
@@ -209,22 +208,12 @@ func (r *readyzOpts) getReadyzHandlerFunc(ctx context.Context) http.HandlerFunc
 
 		defer r.clientPool.Return(etcdClient)
 
-		// Learner and voting members both support the endpoint status call
-		resp, err := etcdClient.Status(ctx, r.targetEndpoint)
-		if err != nil {
-			klog.V(2).Infof("failed to get member endpoint status: %v", err)
-			http.Error(w, fmt.Sprintf("failed to get member endpoint status: %v", err), http.StatusServiceUnavailable)
-			return
-		}
+		timeout, cancel := context.WithTimeout(ctx, 60*time.Second)
```

Comment:
Should this context be derived from the context of the `*http.Request` (`r.Context()`) passed to each handler invocation?